utils/eventBus.ts

```diff
@@ -0,0 +1,33 @@
+type UnkownFunction = (...args: unknown[]) => unknown
+
+class EventBus {
```

Comment:
I remember that we always avoid using the event bus in the codebase. It may make the codebase a little bit hard to manage.

Reply:
yeah I get that
this is what GalleryItem does on 'buy-success' event
```ts
const onNFTBought = () => {
  activeTab.value = tabs.activity
  showCongratsMessage.value = true
}
EventBus.on('buy-success', onNFTBought)
```

1.showCongratsMessage could be moved to Buy.Vue or omitted completely
2. activeTab... I don't have a good idea on how to preserve this behavior without event bus, open to suggestions

Reply:
Please no event-bus it's a Bad pattern, maybe try to tweak it with some composables

Reply:
>  activeTab... I don't have a good idea on how to preserve this behavior without event bus, open to suggestions

One option is using query params on URL to trigger the event.

Reply:
I've dropped the event bus and I'm using the preferences store to "transport" the message

I don't think it's a great solution honestly, mainly because the store holds state and not really event
so it becomes the event recover responsibility to turn the state back to `false`

URL param is the same but worse I think

and there is the option of relaying the event through a composable, but it's the same as event bus just 'dressed up' differently

a composable 'event bus' will look something like:


```ts
export default function useEvent() {
  const events = ref<string[]>([])

  const emitEvent = (eventName: string) => {
    events.value.push(eventName)
  }

  const onEvent = (eventName: string, callback: () => void) => {
    const unwatch = watch(
      () => events.value,
      () => {
        if (events.value.includes(eventName)) {
          callback()
          events.value = events.value.filter(event => event !== eventName)
        }
      },
      { immediate: true }
    )

    return unwatch
  }

  return {
    emitEvent,
    onEvent,
  }
}
```

Reply:
> I've dropped the event bus and I'm using the preferences store to "transport" the message
https://github.com/kodadot/nft-gallery/commit/81651bcdc6422896a16b31c77e569726ba4546f8

IMO the store implementation is better than the previous one. We still do not need an event bus wrapped in the composable.